### ISSUE TITLE
.NET7: Better dotnet paths detection

### ISSUE THREAD

**Issue by brmassa** [FlaxEngine/FlaxEngine#956]
@GoaLitiuM invested a lot of time not only in implementing dotnet7 and porting it to Linux. Congrats congrats congrats!

However, the current `./GenerateProjectFiles.sh` process does not work on Ubuntu. Ubuntu (and probably other non-_rolling-release_ distros) use `DISTRONAME.DISTROVERSION.ARCH` naming scheme. To make it more flexible, I suggest using `System.Runtime.InteropServices` classes (in particular `RuntimeInformation` and `RuntimeEnvironment`) to get information about .NET runtime and sdk.

to make a quick fix, replace line 107:
```csharp
rid = $"{osId}-{arch}";
```

by:
```csharp
rid = System.Runtime.InteropServices.RuntimeInformation.RuntimeIdentifier.ToString();
```

to further embrace this idea, I would suggest replacing ALL win/linux/mac special codes for a unified solution using the mentioned classes, like `System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory()`

**Comment by brmassa** [comment]
I urge the devs to use this quick fix.
It does not compile on Ubuntu (the most popular distro) because it uses a different naming for dotnet package.

**Comment by mafiesto4** [comment]
Sorry for delay. I merged it to `1.6` branch.
